src/sections.rs

```diff
@@ -142,6 +142,10 @@ macro_rules! define_custom_sections {
 define_custom_sections! {
     Name(Vec<NameSubSection>) = "name",
     Producers(Vec<ProducerField>) = "producers",
+    // https://github.com/WebAssembly/tool-conventions/blob/08bacbed/Debugging.md#external-dwarf
+    ExternalDebugInfo(String) = "external_debug_info",
```

Comment:
Since this name doesn't clarify what a `String` is, I think it's reasonable to change it to

```suggestion
    ExternalDebugInfo { url: String } = "external_debug_info",
```

Reply:
> I couldn't get the second syntax working though, I'm probably missing something.

Ah is that what you meant? Hmm maybe my derive macro doesn't support that syntax yet.

Reply:
Hmm it seems to compile fine in other cases. Can you provide more details on why `{ url: String }` didn't work for you - did it fail to compile or lead to invalid results or something else?

Reply:
This is beyond my understanding of rust but I think the macro doesn't currently support that syntax. I'm blindly copying your suggestion and getting:
```
error: no rules expected the token `{`
   --> src/sections.rs:146:23
    |
101 | macro_rules! define_custom_sections {
    | ----------------------------------- when calling this macro
...
146 |     ExternalDebugInfo {url: String} = "external_debug_info",
    |                       ^ no rules expected this token in macro call

error[E0412]: cannot find type `CustomSection` in module `super`
   --> src/sections.rs:409:19
    |
94  | pub struct RawCustomSection {
    | --------------------------- similarly named struct `RawCustomSection` defined here
...
409 |     Custom(super::CustomSection) = 0,
    |                   ^^^^^^^^^^^^^ help: a struct with a similar name exists: `RawCustomSection`


```

Reply:
Oh, so it's not derive, but just this internal macro that doesn't support field syntax. That should be easy to fix. Are you familiar with macros enough to extend `define_custom_sections` or do you want me to do that?

Reply:
Or, you know what, those variants should be rare enough that it doesn't matter that much. If you're okay with `(String)` maybe let's just go with that.

Reply:
I'm alright with it I think. I took a stab at extending the macro but didn't really get anywhere.

Reply:
Ok let's just leave it simple then. It's a bit unfortunate, but I assume if someone wants this section, they know what contents to expect.